app/controllers/users_controller.rb

```diff
@@ -1,6 +1,6 @@
 class UsersController < ApplicationController
 
-  before_action :admin_user,     only: :destroy
+  before_action :admin_user
```

Comment:
Non-admins are still allowed to create a profile (but not a user), but they are restricted from making changes to existing profiles or users.

Reply:
I don't think they could ever do that.  They could only get to the new profile form, but the create method was still protected so it wouldn't go through.  In any case, @lequytra has locked that down so no one can even see the form.